homeassistant/components/nasweb/climate.py

```diff
@@ -0,0 +1,168 @@
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.typing import DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import (
+    BaseCoordinatorEntity,
+    BaseDataUpdateCoordinatorProtocol,
+)
+
+from . import NASwebConfigEntry
+from .const import DOMAIN, STATUS_UPDATE_MAX_TIME_INTERVAL
+
+CLIMATE_TRANSLATION_KEY = "thermostat"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: NASwebConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up Climate platform."""
+    coordinator = config.runtime_data
+    nasweb_thermostat: NASwebThermostat = coordinator.data[KEY_THERMOSTAT]
+    climate = Thermostat(coordinator, nasweb_thermostat)
+    async_add_entities([climate])
+
+
+class Thermostat(ClimateEntity, BaseCoordinatorEntity):
+    """Entity representing NASweb thermostat."""
+
+    _attr_device_class = SensorDeviceClass.TEMPERATURE
+    _attr_translation_key = CLIMATE_TRANSLATION_KEY
+
+    def __init__(
+        self,
+        coordinator: BaseDataUpdateCoordinatorProtocol,
+        nasweb_thermostat: NASwebThermostat,
+    ) -> None:
+        """Initialize Thermostat."""
+        super().__init__(coordinator)
+        self._thermostat = nasweb_thermostat
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._attr_precision = 1.0
+        self._attr_supported_features = ClimateEntityFeature(
+            ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+        )
+        self._attr_available = False
+        self._attr_hvac_modes = [
+            HVACMode.OFF,
+            HVACMode.HEAT,
+            HVACMode.COOL,
+            HVACMode.HEAT_COOL,
+            HVACMode.FAN_ONLY,
+        ]
+        self._attr_name = nasweb_thermostat.name
+        self._attr_target_temperature_step = 1.0
+        self._attr_min_temp = -50
+        self._attr_max_temp = 50
+        self._attr_has_entity_name = True
+        self._attr_should_poll = False
+        self._attr_unique_id = f"{DOMAIN}.{self._thermostat.webio_serial}.thermostat"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._thermostat.webio_serial)}
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """When entity is added to hass."""
+        await super().async_added_to_hass()
+        self._handle_coordinator_update()
+
+    def _set_attr_available(
+        self, entity_last_update: float, available: bool | None
+    ) -> None:
+        if (
+            self.coordinator.last_update is None
+            or time.time() - entity_last_update >= STATUS_UPDATE_MAX_TIME_INTERVAL
+        ):
+            self._attr_available = False
+        else:
+            self._attr_available = available if available is not None else False
```

Comment:
This could be moved to a new class `NaswebEntity`, it's the same for all nasweb entities. OK to do that in a separate PR.